connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+openapi: 3.0.0
+info:
+  description: |
+    Connector Builder Server API
+
+  version: "1.0.0"
+  title: Connector Builder Server API
+  contact:
+    email: contact@airbyte.io
+  license:
+    name: MIT
+    url: "https://opensource.org/licenses/MIT"
+externalDocs:
+  description: Find out more about Connector Builder
+  url: "https://docs.airbyte.com/connector-development/config-based/overview/"
+
+paths:
+  /v1/stream/read:
```

Comment:
I tried a nested grouping here of pages nested inside of slices, with a sliceId and pageId in each nested object.

What do you think of this structure? I thought it made sense because if a user is doing both stream slicing _and_ pagination, then seeing each page broken down by slice makes sense. If they are using only one of those, then there will just be one element in other level, e.g. if they are not using pagination then there will just be a single element in `pages`.

I also wasn't super sure about `sliceId` and `pageId` -- I figured we wanted some way to identify which slices and which pages are which, but I wasn't sure if we have something to use as the "IDs" here. Since they are just strings, maybe we can just put in the URL parameter / header that changes for each slice/page?

Reply:
in practice, stream slices correspond to states, so we could identify them using the state object, for example:
```
{
"repository": "airbytehq/integration-test",
"created_at": "2021-06-29T03:44:45Z"
}
```

we don't currently persist or keep track of the number of pages within a slice, but each page corresponds to a single request/response so we could identify them using a counter

Reply:
Got it, I think that makes sense as a way to differentiate them

Reply:
I decided to just remove the `sliceId` and `pageId` fields from this. I didn't feel like they would add much value, because the slice "state" can be extracted from the results, and the page number is basically just the index of the page in the `pages` array

Reply:
> the slice "state" can be extracted from the results
the slice represents a range ("2021-05-29T03:44:45Z" - "2021-06-29T03:44:45Z"), whereas the output state only represents the end state.
Do we only care about showing the end state here?